Annotations for gene legend in viewer

mwshinn opened the pull request:
This creates labels for the different colours in the gene legend.  It operates by (a) adding a new gene_legend_order_by option, and (b) creating a new column in the gene_colours.csv file.  It sets this new option to be the default.

I'm not sure what a unit test would look like for this, but I confirmed that everything still works when the gene legend is sorted by colour, as is the current default.

There are some genes I am not familiar with in the default gene colour file (not in 73g) so I left them as question marks - these should be fixed before merging.

paulshuker committed:
[extract] [changelog] image chunks no longer too small fix, closes #199

paulshuker committed:
Merge pull request #204 from paulshuker/199-extract-larger-chunk-size

[extract] [changelog] image chunks no longer too small fix, closes #199

mwshinn committed:
Annotations for gene legend in viewer

mwshinn commented:
Forgot to attach the screenshot: 
![DeepinScreenshot_select-area_20250120201451](https://github.com/user-attachments/assets/b264fc1e-dbf1-4cc9-b034-cd3f72662058)

paulshuker commented:
Hey @mwshinn this is great! I am happy to merge this in.
Can you make the same PR into branch `1.0.0-post` please? I like to keep this branch as a staging branch and it already includes (not gene legend) Viewer code changes.
I think I can tell from the code that this new cell_type column is optional, which would be ideal.

mwshinn committed:
Update gene colour labels

mwshinn committed:
Merge changes

mwshinn commented:
Updated and rebased to 1.0.0-post!  Thanks to Izzie for the help with the gene labels.  Also I checked and it did not work as expected when the column was missing, so I fixed that.

paulshuker commented:
Thank you! I think your part is done now, I will get round to merging/testing it soon.
